setup-scripts/skynet-nginx.conf

```diff
@@ -22,28 +30,33 @@ server {
 	}
 
 	location /stats {
+		limit_req zone=stats; # ddos protection: max 10 requests per minute
+
 		proxy_set_header Access-Control-Allow-Origin: *;
 		proxy_set_header User-Agent: Sia-Agent;
 
-    # replace BASE64_AUTHENTICATION with base64 encoded <user>:<password>
+		# replace BASE64_AUTHENTICATION with base64 encoded <user>:<password>
 		# for sia user is empty so it's just :<password>
 		# to generate the passcode use https://www.base64encode.org or any other base64 encoder
 		proxy_set_header Authorization "Basic BASE64_AUTHENTICATION";
 		proxy_pass http://127.0.0.1:9970/skynet/stats;
-  }
+	}
 
 	location /statsdown {
+		limit_req zone=stats; # ddos protection: max 10 requests per minute
+
 		proxy_set_header Access-Control-Allow-Origin: *;
 		proxy_set_header User-Agent: Sia-Agent;
 
-    # replace BASE64_AUTHENTICATION with base64 encoded <user>:<password>
+		# replace BASE64_AUTHENTICATION with base64 encoded <user>:<password>
 		# for sia user is empty so it's just :<password>
 		# to generate the passcode use https://www.base64encode.org or any other base64 encoder
 		proxy_set_header Authorization "Basic BASE64_AUTHENTICATION";
 		proxy_pass http://127.0.0.1:9980/skynet/stats;
-  }
+	}
 
 	location /skynet/skyfile/ {
+		limit_conn uploads 10; # ddos protection: max 10 uploads at a time
```

Comment:
Is this defined per zone? Which I believe in our case is per unique IP? Perhaps a good idea to include it in the comment. If it's per IP I feel these are extremely limiting. Also by default it returns an error most probably, but which one? I think I would probably expect 429 but I'm unsure if that is the default.

Reply:
I think 10 might be a little low. I think Ivo implemented parallel uploads in siac which defaulted to 8, unsure at what he capped that but we should probably make it so the max value there is the default value here? In any case 10 feels a bit restrictive